functions/src/courses/Course.controller.ts

```diff
@@ -0,0 +1,29 @@
+import { Controller, Get, Path, Query, Route } from 'tsoa';
+import { Course } from './Course.model';
+import { CoursesService } from './Course.service';
+
+@Route('courses')
+export class CoursesController extends Controller {
+  /**
+   * Retrieves all the courses available. If query params are passed in, they will be used to filter results.
```

Comment:
My thinking is typically they exist to reduce what's being sent to the client. Given there's like 3400~ entries it would be nice to for the API to slim down the response if requested.

Reply:
Can you describe a use case for this? I see what you're doing now, filter by either subject or code (or both) (ie. get all courses with subject="CSC").

Reply:
Fair point. We'll add it as we need when the use case comes up.